### ISSUE TITLE
calling method from _integrate_forces() doesn't always register

### ISSUE THREAD

**Issue by tommiemartin** [godotengine/godot#43806]
**Godot version:**
3.2.3

**OS/device including version:**
OSX, iOS, Android

Not sure if this is related to the other physics issue but calling methods or signals with in integrate_forces contacts isn't always registering.

This code works majority of the time but for some reason sometimes fails to trigger the objects method.
```
func _integrate_forces(s): 
	velocity = s.get_linear_velocity()
	spin = s.get_angular_velocity()
	step = s.get_step() 
 
		for x in range(s.get_contact_count() ): 
		  var cn = s.get_contact_local_normal(x) 
		  var p = s.get_contact_collider_position(x) 
	  	  var o = s.get_contact_collider_object(x)
 
		  # Push collisions,  
		  if o is KinematicBody:
			if o.has_method("_push"):
				o._push(-cn * step)
#				o.call_deferred("_push",-cn * step) # recommend on discord doesn't help
 
		  #Play sound 
		  if abs(cn.z) >= 0.6:
			  apply_central_impulse(-velocity * step)
			  if o.get_collision_layer_bit(9):
				soundWall.play() # playSoundWallHit()
```

**Comment by Calinou** [comment]
Can you try putting `print` statements in various branches of the code to see which branch isn't being called when it should?

**Comment by tommiemartin** [comment]
Yes, it appears to be in the `s.get_contact_count()` loop.  (contact count set to 3 to account for the floor / possible wall, and `can sleep` disabled. I tried it with this print and sometimes it registering and sometimes I'll be colliding /pushing against the enemy with nothing triggering.

```
		if not o is StaticBody:
			print("Player collided with ", o.get_name() )
```

**Comment by pouleyKetchoupp** [comment]
@tommiemartin Please upload a minimal project to allow contributors to test this issue with your scene setup.

**Comment by tommiemartin** [comment]
Honestly I'm not using Godot anymore and ended up shipping my game with this issue, Be it a problem with physics or problem between chair and screen. I'm not sure if _integrate_forces(s): really gets used that much the way I was doing it. If it helps I can just close this out.

**Comment by pouleyKetchoupp** [comment]
Thanks for your answer! Yes, let's close it for now since we don't have a specific repro case.

For anyone hitting a similar issue, please comment with a minimal project so it can be investigated.

**Comment by valknor** [comment]
Hello,

I have experienced a similar problem, which appears to be restricted to the Bullet Physics Engine. My code is below:

`_integrate_forces(state):

	print(state.get_contact_count())
	if(state.get_contact_count() >= 1):  #this check is needed or it will throw errors 
		local_normal0 = state.get_contact_local_normal(0)`
		
Neither state.get_contact_count, nor state.get_contact_local_normal consistently returns a value other than 0, even when the rigid body is clearly on the ground. I've found that this issue depends on the kind of collision mesh in use, with trimesh collision siblings producing this problem far less often. However on more complex meshes, such as inside cylinders this still sometimes happens. I would like to say using Godot Physics is a workaround, but it introduces other issues with friction and some unusual collision problems.

I am uploading a minimal version of my project which should reproduce these problems for you. Rolling toward the middle (green ramp) consistently returns 0 for the above printout. Controls are pretty standard, WASD, Space for jump, and mouse for camera rotation.

[gammaball.zip](https://github.com/godotengine/godot/files/8112352/gammaball.zip)
